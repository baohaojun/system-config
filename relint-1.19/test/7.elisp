;;; Relint test file 7          -*- emacs-lisp -*-

(defun my-dolist-fun (seq)
  (let ((s ""))
    (dolist (c seq)
      (setq s (concat s (char-to-string c))))
    s))

(defun test-dolist ()
  (looking-at (my-dolist-fun '(?a ?b ?^))))

(defun my-while-fun ()
  (let ((s "")
        (c ?!))
    (while (< c ?&)
      (setq s (concat s (char-to-string c)))
      (setq c (1+ c)))
    s))

(defun test-while ()
  (looking-at (my-while-fun)))

(defun test-mapc ()
  (looking-at (let ((s ""))             ; "[**]"
                (mapc (lambda (x) (setq s (concat s x)))
                      '("[" "*" "*" "]"))
                s)))

(defun test-eval-cl-loop ()
  (looking-at                           ; "!\"#$%"
   (apply 'string (cl-loop for i in (number-sequence ?! ?%) collect i))))

(defun test-catch ()
  (looking-at (catch 'boing "[XX]")))

(defun test-condition-case ()
  (looking-at (condition-case err "[XX]" (error "Y"))))
