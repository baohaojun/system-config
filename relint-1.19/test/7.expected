7.elisp:10:15: In call to looking-at: Unescaped literal `^' (pos 2)
  "ab^"
   ..^
7.elisp:21:15: In call to looking-at: Unescaped literal `$' (pos 3)
  "!\"#$%"
   ....^
7.elisp:24:15: In call to looking-at: Duplicated `*' inside character alternative (pos 2)
  "[**]"
   ..^
7.elisp:31:4: In call to looking-at: Unescaped literal `$' (pos 3)
  "!\"#$%"
   ....^
7.elisp:34:15: In call to looking-at: Duplicated `X' inside character alternative (pos 2)
  "[XX]"
   ..^
7.elisp:37:15: In call to looking-at: Duplicated `X' inside character alternative (pos 2)
  "[XX]"
   ..^
